File: Modul 6/Soal 4/PRAK604-2410817210014-ZAVIER GHANI.py
kode = input("Masukkan kode yang dimiliki oleh Shikamaru: ")
pesan = input("Masukkan pesan yang diterima oleh Shikamaru: ")

if len(kode) != len(pesan):
    print("Panjang kalimat berbeda, pesan palsu")
else:
    bintang = 0
    pagar = 0
    hasil = ""

    for k, p in zip(kode, pesan):
        if k == p:
            hasil += "*"
            bintang += 1
        else:
            hasil += "#"
            pagar += 1

    print(hasil)
    print(f"* = {bintang}")
    print(f"# = {pagar}")

    if bintang >= pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

File: Modul 6/Soal 4/PRAK604-2410817210014-ZAVIERGHANI.c
#include <stdio.h>
#include <string.h>

int main() {
    char kode[100], pesan[100];
    int bintang = 0, pagar = 0;

    printf("Masukkan kode yang dimiliki oleh Shikamaru: ");
    scanf(" %[^\n]", kode);
    printf("Masukkan pesan yang diterima oleh Shikamaru: ");
    scanf(" %[^\n]", pesan);

    if (strlen(kode) != strlen(pesan)) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
        return 0;
    }

    for (int i = 0; i < strlen(kode); i++) {
        if (kode[i] == pesan[i]) {
            printf("*");
            bintang++;
        } else {
            printf("#");
            pagar++;
        }
    }
    printf("\n");

    printf("* = %d\n", bintang);
    printf("# = %d\n", pagar);

    if (bintang >= pagar) {
        printf("Pesan Asli\n");
    } else {
        printf("Pesan Palsu\n");
    }

    return 0;
}
